### PR TITLE
Feature/27 coding standards

### LINKED ISSUE #27: Draft, document, and integrate coding standards
As our team grows, it's necessary to follow specific coding standards. Let's start and iterate together.

- [x] Check on official C# + Unity formatting guidelines
- [x] Check for other Unity projects and what standards they followed
- [ ] Document our initially drafted standards (and/or link to official ones)
- [ ] Integrate these standards into VS and Rider (PoC)
- [ ] Discuss first iteration with team and iterate

Out of scope:
* Changing existing code
* Full blown solution of everything - think of it more like a vertical slice. Especially testing the integration within IDEs. And do not overcomplicate it by adding hundreds of rules

### PR DESCRIPTION
**Pull request by JaXt0r**
# Please review briefly:

* Start Rider and check if the following lines will hint a yellow warning:
* (optional) Test the same warnings with VS22
---
- [ ] 1. ```private bool isTest;``` <-- Needs to be underscored
image
---
- [ ] 2. ```protected bool isTest2;``` <-- Needs to be PascalCase like every non-private field
image
---
- [ ] 3. Braces are needed.
```c#
if (isActiveAndEnabled)
    return;
```
image
image

---
# Out-of-scope:
* I added only ~5 checks. Please look at .editorconfig for details. We can update in the future whenever needed.
* I only linked to "auto-format on save" documentation. I think this is sufficient.

### CODE REVIEW

**Comment by ThomasMountainborn** [comment]
> protected bool isTest2; <-- Needs to be PascalCase like every non-private field

Hmm, I find that a bit weird, since then you still can't immediately tell the difference between a local variable and a member. The difference between private and protected isn't very relevant in the code, but the difference between local and member is. I find it clearest this way:

PublicMember
_privateMember
localVariable

**Comment by JaXt0r** [comment]
> > protected bool isTest2; <-- Needs to be PascalCase like every non-private field
> 
> Hmm, I find that a bit weird, since then you still can't immediately tell the difference between a local variable and a member. The difference between private and protected isn't very relevant in the code, but the difference between local and member is. I find it clearest this way:
> 
> PublicMember _privateMember localVariable

Yep. 100%. I didn't phrase it properly. But meant the same outcome as you said:

public PascalCase;
protected PascalCase;
private _underscoreCase;
Local: camelCase;

File: Docs/development/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+        // This groups data from the custom PlayerStats class in the Inspector.
+        [SerializeField] private PlayerStats _stats;
+
+        // This limits the values to a Range and creates a slider in the Inspector.
+        [Range(0f, 1f)] [SerializeField] private float _rangedStat;
+
+        // A tooltip can replace a comment on a serialized field and do double duty.
+        [Tooltip("This is another statistic for the player.")]
+        [SerializeField] private float _anotherStat;
```
Comment (lmichaelis) [inline Docs/development/style-guide.md]: We should mention Unity-style for inspector fields here (camelCase). Do we adhere to it or do we ignore it and go with `PascalCase` anyways?

**Review by lmichaelis — COMMENTED** [review]
Looks generally good :)

File: Docs/development/style-guide.md (hunk 1 above)
Comment (JaXt0r) [reply Docs/development/style-guide.md]: What do you mean? The example above leverages UnityFields the same as non-Unity fields. What is default and what would change in here?
```[SerializeField] private float _anotherStat;```

Comment (lmichaelis) [reply Docs/development/style-guide.md]: Ah did I misunderstand? Are we going to use private members even for properties set through the inspector?

Comment (JaXt0r) [reply Docs/development/style-guide.md]: I didn't include this as a recommendation. What's your recommendation? Then I can add this information.

Comment (JaXt0r) [reply Docs/development/style-guide.md]: Ok. I added another hint at this. Anything else needed? If not, feel free to accept the PR.

```c#
        // Use [SerializeField] attribute if you want to display a private field in Inspector.
        // Try to use private Fields for Components/MonoBehaviours, if you don't need them to be public.
        // Booleans ask a question that can be answered true or false.
        [SerializeField] private bool _isPlayerDead;
```

**Review by lmichaelis — APPROVED** [review]

**Pull request merged**